client/modules/IDE/components/FileTypeIcon.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { IoLogoHtml5, IoLogoCss3, IoLogoJavascript } from 'react-icons/io';
+import { TbFileTypeXml, TbTxt, TbCsv } from 'react-icons/tb';
+import { VscJson } from 'react-icons/vsc';
+import FileIcon from '../../../images/file.svg';
+
+export default function FileTypeIcon({ fileExtension }) {
+  switch (fileExtension) {
```

Comment:
This is a great starting point!

As this evolves we'll figure out what makes sense code-wise.  `switch` definitely works.  A dictionary object could work.  That is, `const icons = {   html: IoLogoHtml5,  /// etc. }`.  If we want to use the same icon for more than one extension then `if else` statements might make sense.  Like `if (/jsx?/i.test(fileExtension)) { return <IoLogoJavascript /> }` to match `.js` and `.jsx`.

Reply:
>  If we want to use the same icon for more than one extension then if else statements might make sense. Like if (/jsx?/i.test(fileExtension)) { return <IoLogoJavascript /> } to match .js and .jsx.

In javascript, similar code can be assigned multiple cases can be assigned to multiple case like this:

```
 switch ("1") {
  case "1":
  case "2":
    console.log("test1");
    break;
  case "3":
    console.log("test2");
}
```
So I think switch overall is the best way.

Reply:
Hi @linapaiste. Sorry for the ping but it's been 5 days since I have heard anything from you. 

I think this feature is close to being merged. We only need to decide on the icon pack. I am very excited for it to be merged as it's the first time I'm contributing a feature. 

Please write back. Regards.

Reply:
Hi @lindapaiste. Sorry for the ping but it's been 5 days since I have heard anything from you. 

I think this feature is close to being merged. We only need to decide on the icon pack. I am very excited for it to be merged as it's the first time I'm contributing a feature. 

Please write back. Regards.

Reply:
Hi @lindapaiste. Sorry for the ping but it's been 5 days since I have heard anything from you. 

I think this feature is close to being merged. We only need to decide on the icon pack. I am very excited for it to be merged as it's the first time I'm contributing a feature. 

Please write back. Regards.